README.md

```diff
@@ -1,7 +1,428 @@
+    "data": {
+        "name": "Breadpudding",
+        "color": "B3DCF5"
+    },
+    "auth": "google",
+    "token": "..."
+}
+```
+
+#### status
+
+This event is used to request a status update from the server. The `data` for this type is an empty object.
+
+Example:
+```JSON
+{
+    "type": "status",
+    "data": {},
+    "auth": "google",
+    "token": "..."
+}
+```
+
+## Commands
+
+Commands are actions performed by users using messages prefixed with an exclamation point.
+
+### ban
+
+Usage: `!ban <username>`
+
+Summary: Bans a user from the chat
+
+Minimum Authentication Level: *Unknown*
+
+### help
+
+Aliases: commands
+
+Usage: `!help`
+
+Summary: Lists all of the commands that the user has access to
+
+Minimum Authentication Level: `0`
+
+### info
+
+Usage: `!info`
+
+Summary: Displays information about the server
+
+Minimum Authentication Level: `0`
+
+### ipban
+
+Usage: `!ipban <address>`
+
+Summary: Blocks a specific IP address from sending messages
+
+Minimum Authentication Level: *Unknown*
+
+### mod
+
+Usage: `!mod <username>`
+
+Summary: Promotes a user to a moderator
+
+Minimum Authentication Level: *Unknown*
+
+### rm
+
+Usage: `!rm <message id>`
+
+Summary: Deletes a message from the chat
+
+Minimum Authentication Level: *Unknown*
+
+### time
+
+Usage: `!time`
+
+Summary: Displays the current time on the server
+
+Minimum Authentication Level: `0`
+
+### timer
+
+Usage: `!timer <start/check/stop> <name>`
+
+Summary: Creates, checks, or deletes a timer on the server
+
+Minimum Authentication Level: `0`
+
+### unban
+
+Usage: `!unban <username>`
+
+Summary: Reverts an existing ban on a user
+
+Minimum Authentication Level: *Unknown*
+
+### unmod
+
+Usage: `!unmod <username>`
+
+Summary: Demotes a user to an unprivileged member
+
+Minimum Authentication Level: *Unknown*
+
+## Message Formatting
+
+*TODO*
+
+### Emoji
+
+The emoji system is still very unknown, but it seems that they are pulled from an array or table.
+
+https://stream.mattkc.com/chat/emotes/
+
+![ Status | Method | Domain | File | Initiator | Type | Transfered | Size | 0ms
+200 | GET | (lock icon) stream.mattkc.com | /chat/emotes/ | jquery-3.6.1.min.js:2 (xhr) | html | 1.65 kB | 7.43kB | 231ms ](images/protocolemoji.png)
+### Mentions
+
+*TODO*
+
+### URLs
+
+*TODO*
+
+## Authentication Levels
+
+The following is a list of the currently known authentication levels:
+
+| Level | Meaning                              |
+| ----- | ------------------------------------ |
+| `0`   | An unprivileged member of the server |
+| `50`  | Server moderator                     |
+| `100` | The owner of the server              |
+
+### Permissions
+
+| Level | `!ban`    | `!ipban`  | `!mod`    | `!rm` | `!unban`  | `!unmod`  |
+| ----- | --------- | --------- | --------- | ----- | --------- | --------- |
+| `0`   | No        | No        | No        | No    | No        | No        |
+| `50`  | *Unknown* | *Unknown* | *Unknown* | Yes   | *Unknown* | *Unknown* |
```

Comment:
We should probably get @janLuna to confirm which commands they have access to and if we're missing any.